RIPS/rip-7712.md

```diff
@@ -39,36 +39,47 @@ with ERC-4337, as all ERC-4337 Smart Contract Accounts already use the solution
 
 ## Specification
 
+### Constants
+
+| Name                     | Value |
+|--------------------------|-------|
+| AA_BASE_GAS_COST         | 15000 |
+| AA_BASE_GAS_COST_RIP7712 | 10000 |
+| AA_NONCE_MANAGER         | tbd   |
+
 ### Non-sequential nonce support
 
-We propose an introduction of a separate 2-dimensional `nonce` value used when contracts initiate a transaction.
 
-The `nonce` parameter of the transaction is to be interpreted as `uint192 key || uint64 seq` value.
-The contract account nonce is then defined as a mapping `address account => uint192 key => uint64 seq`.
+We propose an introduction of `nonceKey` to provide a second dimension
+for the RIP-7560 transaction's `nonce` parameter.
+
+The two-dimensional nonce value of the transaction is represented as `uint192 nonceKey || uint64 nonceSequence` value.
+The contract account nonce is then defined as a mapping `address account => uint192 nonceKey => uint64 nonceSequence`.
 This approach guarantees unique transaction nonce and hash but removes the requirement of nonces being sequential
 numbers.
 
-This `nonce` is exposed to the EVM in a `NonceManager` pre-deployed contract located at the `AA_NONCE_MANAGER` address.
+This two-dimensional nonce mechanism is exposed to the EVM in a `NonceManager` pre-deployed contract
+located at the `AA_NONCE_MANAGER` address.
 
-The `nonce` is [validated and incremented](#nonce-validation-frame) on-chain before the rest of the validation code.
+The two-dimensional nonce is [validated and incremented](#nonce-validation-frame) on-chain
+as part of a `AA_TX_TYPE` transaction validation before the rest of the validation code.
 
-### The old `nonce` account parameter use
+### Nonce validation
 
-The old 64-bit `nonce` account parameter is used when the `key` of the `AA_TX_TYPE` transaction's `nonce` is set to 0.
-Notice that this also prevents the previously signed transactions from becoming valid again.
+The first validation on a transaction is of its nonce.
```

Comment:
```suggestion
```

Reply:
Removed this statement, replaced with something more legible.